addons/mail/models/mail_alias_domain.py

```diff
@@ -0,0 +1,99 @@
+    _description = "Email Domain"
+    _order = 'sequence ASC, id DESC'
+    _rec_name = 'name'
+
+    name = fields.Char('Name', required=True)
+    company_ids = fields.One2many(
+        'res.company', 'alias_domain_id', string='Used In')
+    sequence = fields.Integer(default=10)
+    bounce = fields.Char('Bounce Alias', default='bounce', required=True)
+    bounce_email = fields.Char('Bounce Email', compute='_compute_bounce_email')
+    catchall = fields.Char('Catchall Alias', default='catchall', required=True)
+    catchall_email = fields.Char('Catchall Email', compute='_compute_catchall_email')
+
+    @api.depends('bounce', 'name')
+    def _compute_bounce_email(self):
+        self.bounce_email = ''
+        for domain in self.filtered('bounce'):
+            domain.bounce_email = f'{domain.bounce}@{domain.name}'
+
+    @api.depends('catchall', 'name')
+    def _compute_catchall_email(self):
+        self.catchall_email = ''
+        for domain in self.filtered('catchall'):
+            domain.catchall_email = f'{domain.catchall}@{domain.name}'
+
+    _sql_constraints = [
+        (
+            'code_company_uniq',
+            'UNIQUE(name)',
+            'Alias domain name should be unique'
+        ),
+    ]
+
+    @api.constrains('bounce', 'catchall')
+    def _check_bounce_catchall_uniqueness(self):
+        names = self.filtered('bounce').mapped('bounce') + self.filtered('catchall').mapped('catchall')
+        if not names:
+            return
+
+        existing = self.env['mail.alias'].search(
+            [('alias_name', 'in', list(set(names)))],
+            limit=1,
+        )
+        if existing.alias_parent_model_id and existing.alias_parent_thread_id:
+            # If parent model and parent thread ID both are set, display document name also in the warning
+            document_name = self.env[existing.alias_parent_model_id.model].sudo().browse(existing.alias_parent_thread_id).display_name
+            raise exceptions.ValidationError(
+                _("'%(matching_alias_name)s' is already used by the %(document_name)s %(model_name)s. Choose another alias or change it on the other document.",
+                  matching_alias_name=existing.display_name,
+                  document_name=document_name,
+                  model_name=existing.alias_parent_model_id.name)
+                    )
+        if existing:
+            raise exceptions.ValidationError(
+                _("'%(matching_alias_name)s' is already linked with %(alias_model_name)s. Choose another alias or change it on the linked model.",
+                  matching_alias_name=existing.display_name,
+                  alias_model_name=existing.alias_model_id.name)
+            )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        """ Sanitize bounce / catchall """
+        for vals in vals_list:
+            if vals.get('bounce'):
+                vals['bounce'] = self.env['mail.alias']._sanitize_alias_name(vals['bounce'])
```

Comment:
A bit strange as if you encode ééèèè as alias then it will turn into False and you'll end-up with an error mentioning that bounce is required.

But I guess we want to simplify and not craft custom validation + error messages, which is fine.

EDIT: Looking at tests, this comment might be wrong, I think we replace with "e", a bit hard to follow.